crates/stc_ts_file_analyzer/src/analyzer/types/mod.rs

```diff
@@ -107,6 +107,47 @@ impl Analyzer<'_, '_> {
             | Type::Param(_)
             | Type::Module(_)
             | Type::Tpl(..) => return Ok(ty),
+            Type::Interface(Interface {
+                span,
+                name,
+                type_params,
+                extends,
+                body,
+                metadata,
+            }) => {
+                for extend in extends {
+                    for type_arg in extend.type_args.as_ref() {
+                        for param in type_arg.to_owned().params {
+                            match param {
+                                Type::Array(array) => return Ok(Cow::Owned(Type::Array(array))),
```

Comment:
I tried early return here to see what makes difference but it does not affect the result of `./scripts/test.sh arityAndOrderCompatibility` at all

is `./scripts/test.sh arityAndOrderCompatibility` correct to test?

Reply:
You should add `Type::Interface` to https://github.com/dudykr/stc/blob/d6dc481fd29c9e6e93e268dbd4f11a7b3883a6d2/crates/stc_ts_file_analyzer/src/analyzer/assign/mod.rs#L512

Reply:
The command itself is correct

Reply:
```
$ ./scripts/test.sh arityAndOrderCompatibility

2 unmatched errors out of 17 errors. Got 5 extra errors.
Wanted: [RefError { line: 19, column: 5, code: "TS2741" }, RefError { line: 22, column: 5, code: "TS2741" }]
Unwanted: [(19, "TS2322"), (22, "TS2322"), (30, "TS2322"), (34, "TS2322"), (35, "TS2322")]
```

I made it to remove `(16, "TS2461")`.
Should i remove all the other 5 extra errors In this issue? 😢